test/e2e_node/util.go

```diff
@@ -394,19 +394,24 @@ func getCRIClient() (internalapi.RuntimeService, internalapi.ImageManagerService
 }
 
 // TODO: Find a uniform way to deal with systemctl/initctl/service operations. #34494
-func findRunningKubletServiceName() string {
-	stdout, err := exec.Command("sudo", "systemctl", "list-units", "*kubelet*", "--state=running").CombinedOutput()
-	framework.ExpectNoError(err)
-	regex := regexp.MustCompile("(kubelet-\\w+)")
-	matches := regex.FindStringSubmatch(string(stdout))
+func findRunningKubeletServiceName() string {
+	stdout, matches := kubeletRunningStatus()
 	framework.ExpectNotEqual(len(matches), 0, "Found more than one kubelet service running: %q", stdout)
 	kubeletServiceName := matches[0]
 	framework.Logf("Get running kubelet with systemctl: %v, %v", string(stdout), kubeletServiceName)
 	return kubeletServiceName
 }
 
+func kubeletRunningStatus() ([]byte, []string) {
+	stdout, err := exec.Command("sudo", "systemctl", "list-units", "*kubelet*", "--state=running").CombinedOutput()
```

Comment:
nit: Can you add a comment with a sample output from this command? Maybe include the systemctl version in the comment. Without the output, it's very hard to review the regex.

Reply:
+1.

Reply:
I'm testing on COS image from `make test-e2e-node`

```
test-cos-beta-85-13310-1041-1 /tmp/node-e2e-20201203T012528 $ systemctl --version
systemd 239
+PAM +AUDIT -SELINUX +IMA +APPARMOR +SMACK -SYSVINIT +UTMP -LIBCRYPTSETUP +GCRYPT -GNUTLS -ACL -XZ +LZ4 +SECCOMP +BLKID -ELFUTILS +KMOD -IDN2 -IDN -PCRE2 default-hierarchy=hybrid
```

This is the result of list-units AFTER the test, the regex meant to get the Kubelet service prefix:  

```
test-cos-beta-85-13310-1041-1 /tmp/node-e2e-20201203T012528 $ systemctl list-units '*kubelet*'
UNIT                                                               LOAD   ACTIVE SUB     DESCRIPTION
home-kubernetes-containerized_mounter-rootfs-var-lib-kubelet.mount loaded active mounted /home/kubernetes/containerized_mounter/rootfs/var/lib/kubelet
kubelet-20201203T012528.service                                    loaded active exited  /tmp/node-e2e-20201203T012528/kubelet --kubeconfig /tmp/node-e2e-20201203T012528/kubeconfig --root-dir /var/lib/kubelet --v 4 --logtostderr --dynamic-config-dir /tmp/node-e2e-20201203T012528/dynamic-kubelet-config --network-plugin=kubenet --cni-bin-dir /tmp/node-e2e-20201203

LOAD   = Reflects whether the unit definition was properly loaded.
ACTIVE = The high-level unit activation state, i.e. generalization of SUB.
SUB    = The low-level unit activation state, values depend on unit type.

2 loaded units listed. Pass --all to see loaded but inactive units, too.
To show all installed unit files use 'systemctl list-unit-files'.
```